### PR TITLE
Update plugins, dependencies and Gradle versions

### LINKED ISSUE #114: [BUG] Fix the publish step to the Gradle Plugins Portal and housekeeping
**Describe the bug**
Release 2.4.0 failed when attempting to publish to the Gradle Plugins Portal:
https://app.circleci.com/pipelines/github/sonatype-nexus-community/scan-gradle-plugin/574/workflows/252f210b-f6eb-4ea5-9c88-a482d3b6bba2/jobs/3218
```
Invalid publication, you have different artifacts with the same hash: (jar: and jar: and jar:)
```

Checking the plugin `com.gradle.plugin-publish`, the next version from the one used here, there is an improvement for checking the artifacts. Also, the latest version fails the building due to a test dependency using a vulnerable Log4J version, which will no longer be acceptable in the plugins portal: https://blog.gradle.org/log4j-vulnerability

So, in general we seem to need an update on all plugins and dependencies.

### PR DESCRIPTION
**Pull request by guillermo-varela**
Brings up to date plugins, dependencies and Gradle versions used.

Also removes a test dependency which was using a vulnerable Log4J version and was no longer updated.

Additionally, a small change was made for the publishing to avoid trying to send an artifact twice to the Gradle Plugins Portal, which is possibly why release for 2.4.0 failed initially.

It relates to the following issue #s:
* Fixes #114

cc @bhamail / @DarthHater / @guillermo-varela / @shaikhu

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -129,9 +133,6 @@ publishing {
       // avoid infinite CircleCI builds triggered by release commits
       preCommitText = '[skip ci] '
       buildTasks = ['assemble']
-      git {
-        requireBranch = 'main'
```
Comment (guillermo-varela) [inline build.gradle:133]: Default is already `main`
https://github.com/researchgate/gradle-release/blob/main/README.md#configuration
Also, the newest version for the plugin fails as that property is now final.

File: build.gradle [hunk 2]
```diff
@@ -160,10 +161,6 @@ publishing {
     }
 
     publications {
-      shadow(MavenPublication) { publication ->
-        project.shadow.component(publication)
-      }
```
Comment (guillermo-varela) [inline build.gradle:165]: Apparently this is already done automatically by the release plugin:
https://github.com/johnrengelman/shadow/blob/129549107b6c26ae7000d427c0d60652ab6af788/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/ShadowExtension.groovy#L23-L27

File: build.gradle [hunk 3]
```diff
@@ -91,12 +91,16 @@ dependencies {
   testImplementation "commons-io:commons-io:$commonsIoVersion"
   testImplementation "org.powermock:powermock-module-junit4:$powermockVersion"
   testImplementation "org.powermock:powermock-api-mockito2:$powermockVersion"
-  testImplementation "com.github.ardenliu:arden-file:$ardenFileVersion"
+}
+
+// Remove the gradleApi so it isn't merged into the jar file when applying 'shadow'
+configurations.named(JavaPlugin.API_CONFIGURATION_NAME) {
+  dependencies.remove(project.dependencies.gradleApi())
```
Comment (guillermo-varela) [inline build.gradle:98]: Recommended at: https://github.com/johnrengelman/shadow/blob/c6d25fac31bd4bb8e783ab074156703aee2cdf7f/build.gradle#L23-L26

**Review by shaikhu — APPROVED** [review]
+1 LGTM

**Pull request merged**